app/services/catalog/order_item_sanitized_parameters.rb

```diff
@@ -40,29 +40,16 @@ def service_plan_schema
 
     def sanitized_parameters
       svc_params = ActiveSupport::HashWithIndifferentAccess.new(service_parameters)
-      if service_plan_schema[:properties].present?
-        service_plan_schema[:properties].each_with_object({}) do |(key, attrs), result|
-          value = hide?(key, attrs) ? MASKED_VALUE : svc_params[key]
-          result[attrs[:title]] = value
-        end
-      else
-        service_plan_schema[:schema][:fields].each_with_object({}) do |(field), result|
-          value = hide_ansible?(field) ? MASKED_VALUE : svc_params[field[:name]]
-          result[field[:name]] = value
-        end
+      service_plan_schema[:schema][:fields].each_with_object({}) do |field, result|
```

Comment:
In the near future we will need to deal with empty service planes, think JobTemplates/Workflows without surveys, so we should probably address that now.

Suggest putting `service_plan_schema[:schema][:fields]` into a method so we can detect and return an empty hash.

Maybe something like:
`service_plan_schema.dig(:schema, :fields) || {}`

Reply:
Good catch, I hadn't thought about that. I added a `fields` method that does exactly what you put in the last line.